### PR TITLE
DEV: Convert core components to native class syntax (batch 1)

### PR DESCRIPTION
**Pull request by davidtaylorhq**
Changes made using the ember-native-class-codemod, plus some manual tweaks

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/modal/share-topic.js [hunk 1]
```diff
@@ -112,11 +115,5 @@ const ShareTopicModal = Component.extend(bufferedProperty("invite"), {
     const topicController = getOwner(this).lookup("controller:topic");
     topicController.actions.replyAsNewTopic.call(topicController, post);
     this.closeModal();
-  },
-});
-
-ShareTopicModal.reopenClass({
-  modalClass: "share-topic-modal",
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/components/modal/share-topic.js:119]: so this wasn't used at all? 😃

File: app/assets/javascripts/discourse/app/components/anonymous-topic-footer-buttons.js [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
+export default class AnonymousTopicFooterButtons extends Component {
+  elementId = "topic-footer-buttons";
+  role = "region";
 
-  attributeBindings: ["role"],
-
-  role: "region",
-
-  allButtons: getTopicFooterButtons(),
+  @getTopicFooterButtons() allButtons;
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/components/anonymous-topic-footer-buttons.js:11]: I think non-decorator version makes more sense here?

```suggestion
  allButtons = getTopicFooterButtons();
```

**Review by CvX — APPROVED** [review]

File: app/assets/javascripts/discourse/app/components/modal/share-topic.js (hunk 1 above)
Comment (davidtaylorhq) [reply app/assets/javascripts/discourse/app/components/modal/share-topic.js:119]: Not as far as I can see. It's a 'static property' - very strange.

I suspect (but haven't checked) it was used by the old modal system.

Comment (davidtaylorhq) [reply app/assets/javascripts/discourse/app/components/modal/share-topic.js:119]: Now, it's in the template

https://github.com/discourse/discourse/blob/70ea3f8fcd2959a1435fb100afd0e53872811829/app/assets/javascripts/discourse/app/components/modal/share-topic.hbs#L11C10-L11C10

**Pull request merged**